packages/auth/src/signature/middleware.test.ts

```diff
@@ -401,10 +401,9 @@ describe('Signature Service', (): void => {
       scope.isDone()
     })
 
-    test('middleware succeeds if BYPASS_SIGNATURE_VALIDATION is set', async (): Promise<void> => {
-      deps = await initIocContainer(Config)
+    test('middleware succeeds if BYPASS_SIGNATURE_VALIDATION is true with bad signature', async (): Promise<void> => {
+      const defaultByPassSignatureValidation = Config.bypassSignatureValidation
```

Comment:
The `Config` is a global to the tests and rather than doing a reset each time on the const I'm saving the state of the variable changed and then resetting at the end of the test. I can't recall how Jest treats failed tests (does it run all the way to the end of the test?) so this approach may need to be re-thought.

Reply:
Even with resetting this on `beforeEach` I wasn't able to reset the Config to the default state at the start of the test, so I've reset the state within the test, although I'm still not convinced this is the right way to do this

Reply:
Adding this to the `beforeEach` (I tried the one inside `describe('Signature middleware'`) appears to work
```
      const config = await appContainer.container.use('config')
      config.bypassSignatureValidation = false
```

Reply:
It works, @wilsonianb  but that assumes that the default state is `false` (A reasonable assumption, but not always correct, as Config is set from `process.env` or may have been modified?). 

I'm fine with it, but realized that I might not understand how that `container.use` system works at all, nor do I get the relationship between it and the `Config` import at the top.

It's probably best for you to push any commits changing that because of my lack of understanding.

Reply:
>I might not understand how that `container.use` system works at all

Does anyone really, though? :sweat_smile: 
The imported `Config` is used to construct the container:
https://github.com/interledger/rafiki/blob/c8d42f60386d1c5a0e387c4e8bd30d6f74846f41/packages/auth/src/signature/middleware.test.ts#L44-L45
which is eventually the `ctx.container` in the middleware.
I don't _think_ subsequent changes made to `Config` would be reflected in the container.

You can copy or cherry-pick this: https://github.com/wilsonianb/rafiki/commit/9bf207af6e3895e0391fc8beaa8c819fe21f0433

Reply:
I incorporated your changes @wilsonianb in commit 1ee50db5f31ff69cb55cf191926a48c4f7e36b66 

I then realized there was a cleaner way of allowing tests with multiple configs, so made a subsequent commit to allow the general case which backed out many of these changes. It seems to be working for me and is cleaner, let me know what you think.